packages/itwinui-react/src/utils/providers/ScopeProvider.tsx

```diff
@@ -0,0 +1,65 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import * as React from 'react';
+import { createStore, useAtomValue, useSetAtom } from 'jotai';
+import type { Atom, WritableAtom } from 'jotai';
+
+const ScopeContext = React.createContext({
+  store: createStore(),
+  parentStore: null as null | ReturnType<typeof createStore>,
+});
+
+/**
+ * Provider that creates a fresh, isolated jotai store for its children
+ *
+ * Should be used with `useScopedAtom` and/or `useScopedSetAtom`.
+ *
+ * @private
+ */
+export const ScopeProvider = ({ children }: { children: React.ReactNode }) => {
+  const store = React.useMemo(() => createStore(), []);
+  const parentStore = React.useContext(ScopeContext).store;
+
+  return (
+    <ScopeContext.Provider
+      value={React.useMemo(
+        () => ({ store, parentStore }),
+        [store, parentStore],
+      )}
+    >
+      {children}
+    </ScopeContext.Provider>
+  );
+};
+
+/**
+ * Wrapper over `useAtom` that uses the store from the nearest `ScopeProvider`.
+ *
+ * If the atom is not set in the current store, it will look for the atom in the parent store.
```

Comment:
good point, I changed it so that it actually `set`s the value instead of only returning it. [`9a66baf`](https://github.com/iTwin/iTwinUI/pull/2011/commits/9a66baf63bcbc0750f3e88c981b63cdfa11f70cb)

Reply:
Pushed another change to make it more clear that only initial value is inherited + manually synchronize future updates. [`a1e2845`](https://github.com/iTwin/iTwinUI/pull/2011/commits/a1e2845d7745714920c0e625af218bca45a239e5)

While the future updates are not super important for this particular PR (because the inherited portal container shouldn't change), `useScopedAtom` is a general utility so this pattern might be important in the future.